Inferences/ForwardDemodulation.hpp

```diff
@@ -36,6 +36,8 @@ class ForwardDemodulation
   void detach() override;
   bool perform(Clause* cl, Clause*& replacement, ClauseIterator& premises) override = 0;
 protected:
+  template<bool> struct Applicator;
+
```

Comment:
Why is this necessary? I think the `Applicator` is only used inside `ForwardDemodulation.cpp`.